src/main/java/calculator/DelimiterParser.java

```diff
@@ -0,0 +1,54 @@
+package calculator;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+public class DelimiterParser {
+
+    private static final String CUSTOM_DELIMITER_PREFIX = "//"; // 커스텀 구분자 접두사
+    private static final String CUSTOM_DELIMITER_SUFFIX = "\\n"; // 커스텀 구분자 접미사
+
+    public String[] parseNumbers(String input) {
+        List<String> delimiterList = new ArrayList<>(Arrays.asList(",", ":"));
+        String numbersPart = input; // 숫자 부분
+
+        if (isEndsWithNumber(input)) {
+            throw new IllegalArgumentException("구분자 뒤에 숫자가 없습니다: " + input);
+        }
```

Comment:
메서드명과 실제 역할이 달라 보이네요!
```suggestion
        if (doesNotEndWithNumber(input)) {
            throw new IllegalArgumentException("구분자 뒤에 숫자가 없습니다: " + input);
        }
```